roles/zabbix_proxy/tasks/main.yml

```diff
@@ -143,10 +143,54 @@
     tls_psk: "{{ zabbix_proxy_tlspsk_secret | default(omit) }}"
     tls_psk_identity: "{{ zabbix_proxy_tlspskidentity | default(omit) }}"
     tls_subject: "{{ zabbix_proxy_tls_subject | default(omit) }}"
-    tls_connect: "{{ zabbix_proxy_tls_config[zabbix_proxy_tlsaccept | default('no_encryption')] }}"
-    tls_accept: "{{ zabbix_proxy_tls_config[zabbix_proxy_tlsconnect | default('no_encryption')] }}"
+    tls_accept: "{{ zabbix_proxy_tls_config[zabbix_proxy_tlsaccept | default('no_encryption')] }}"
+    tls_connect: "{{ zabbix_proxy_tls_config[zabbix_proxy_tlsconnect | default('no_encryption')] }}"
   when:
     - zabbix_api_create_proxy | bool
+    - zabbix_proxy_version is version('7.0', '<')
+  delegate_to: "{{ zabbix_api_server_host }}"
+  tags:
+    - api
+
+- name: Ensure proxy definition is up-to-date (added/updated/removed) for Zabbix >= 7.0
+  vars:
+    gather_facts: false
+    ansible_user: "{{ zabbix_api_login_user }}"
+    ansible_network_os: community.zabbix.zabbix
+    ansible_connection: httpapi
+    ansible_httpapi_use_ssl: "{{ zabbix_api_use_ssl }}"
+    http_login_user: "{{ zabbix_api_http_user | default(-42) }}"
+    http_login_password: "{{ zabbix_api_http_password | default(-42) }}"
+  community.zabbix.zabbix_proxy:
+    state: "{{ zabbix_proxy_state }}"
+    operating_mode: "{{ zabbix_proxy_status }}"
+    proxy_name: "{{ zabbix_proxy_name }}"
```

Comment:
The question for the above scenario relates to whether we should atleast provide a default for `zabbix_proxy_name` in `defaults/main.yml`  or if it should be `zabbix_proxy_hostname` here.